Framework/DataHandling/test/SaveNXSPETest.h

```diff
@@ -133,7 +227,68 @@ class SaveNXSPETest : public CxxTest::TestSuite {
       Poco::File(outputFile).remove();
   }
 
+  void xestSaveNXSPE_extractIndirectDetectorsEn() {
+    // TODO: for a future, I do not know how to set energy for a detector
+    // and modern Indirect have same EI anyway
+    const int nhist(3), nx(10);
+    MatrixWorkspace_sptr input = makeWorkspace(nhist, nx);
+
+    std::vector<double> EiValues(nhist);
+    for (int i = 0; i < nhist; i++)
+      EiValues[i] = double(i + 2);
+
+    input = makeWS_indirect(input, EiValues);
+
+    auto loadedData = saveAndReloadWorkspace(input, 10, false);
+    auto detEi = loadedData.get<3>();
+
+    for (int i = 0; i < nhist; i++) {
+      TS_ASSERT_DELTA(EiValues[i], detEi[i], 1.e-8);
+    }
+  }
+  void xest_saveIndirectDetectorsEn() {
+    // TODO: for a future, I do not know how to set energy for a detector
+    // and ISIS Indirect instruments have same EI anyway
```

Comment:
It might be worth asking about this in the tech-qa channel (assuming you haven't aleady). If no luck, then I would say it would be preferred to remove these unused tests completely until we can do a full implementation of them

Reply:
@robertapplin I think in that case we should just remove them and leave it for a later date in case this becomes relevant for indirect instruments.

Reply:
I would keep it as there is code which should be tested by this test. I've tested it manually on working instrument but there are no other traces on what should be done. 

To be fair, this is small side job to satisfy colleagues and I can not spend a lot of time dealing with it. If doing it properly, one needs to find the appropriate method or make mutable method to set detector reduction parameters similar to the one used to get them. Unfortunately I do not have time to do it properly without direct request. 

The thought about getEfixed may be reasonable one. I will have a look.

Reply:
@robertapplin  -- I do not understand this love of hiding incomplete working code. 

When I was looking at unfamiliar saveNXSPE code, I've found number of TODO centences. When these to-do were coinciding with my though and my task, I was able to make some changes -- I believe useful. 

Now I was thinking on how to test the code which is there. Yes I can not complete this work due to different priorities but somebody who would need to complete it, will look at it and will see how I was thinking. This may be useful for finishing or he/she would disagree and reject it but he/she will benifit from what I did. 

Who will benifit from removing this code? Why whould it be better not to keep it there? I do not understand.

Reply:
I would argue that while this code might build, it is not "working" code because the test does not run and pass (i.e. the test name is marked with an `x` at the start so is not being run)

I think there are benefits and downsides to TODO comments. This article gives a good balance https://www.osedea.com/insight/should-you-put-todos-in-the-source-code

My comment is not about the use of TODO, but instead about avoiding adding incomplete code into the codebase until a full implementation is ready. Code that isn't yet functional or complete can introduce confusion, increase technical debt, and complicate future maintenance.

I believe in the principle that "there's nothing more permanent than a temporary solution." Once code is in the repository, there's a risk it may never be revisited or properly implemented, leading to potential issues down the road

Reply:
This is working code and I have tested it manually. I can not add automated test for it because it is not my current priority. 
The best way of maintaining consistent code base is of course not to add any features to it at all. 

It is very good article about TODO in the code and the TODO here are added completely in accordance with these guidlines. 

I can not complete this code because there are no instrument requesting it in its full capacity. But I know that the instrument which will request it is under development (MUSHRUM). I may be asked to provide support for this instrument in Mantid so these changes will be extreamly useful to me. I am also believe that if somebody else is requested to provide support for this instrument, these changes will be useful to them.

There is also big urge to not to use Mantid for supporting new features at all. The reason for that is higlighted by this work: I have spent 1 day of adding useful feature requested by user and now spending week explaining why I have added it and why I am adding it in this particular way. I can not spend any more time on this.

I had a look at `getEFixed` in widgets. To my opinions it should not be there at all, as there are plenty of alternatives in Workspaces and Kernel etc. And I do not think one should introduce dependency  of `DataHandlers` from `qt\widgets`

But `getEFixed` in widgets adds some specific features (`micca`) which I am not familiar with, and the code, calculating `Efixed` for indirect instruments in `SaveNXSPE` adds other feature (multiple detectors energies) which will probably not supported by widgets. 
Reconciling these differences goes far beyound of this ticket and needs discussion between all interested parties which I am not familiar with. 

@SilkeSchomann  -- I can not spend any more time on this work. Could you, please, finish this job. Direct and indirect instruments in ISIS rely on this result. Thank you.